src/base/init/init.c

```diff
@@ -394,9 +404,9 @@ void low_mem_init(void)
   assert(ptr2);
   if (config.dpmi) {
     void *dptr = smalloc_aligned_topdown(&main_pool, ptr2, PAGE_SIZE,
-	dpmi_mem_size());
+	ptr2 - MEM_BASE32(config.dpmi_base));
```

Comment:
That would be OK only if you have
the pool space very "tight". Since you
account LOWMEM_SIZE twice, this
gives the wrong result.
Please use smalloc_fixed() instead.